### PR TITLE
feat(replays): Record analytics on replay details pageload

### PR DESCRIPTION
**Pull request by ryan953**
Enabled analytics debugging by setting `DEBUG_ANALYTICS=1` in localStorage

That debugging setting shows, in the console, the logs to emit:

Screen Shot 2022-08-31 at 9 43 39 PM

Related to #38315

### CODE REVIEW

File: static/app/utils/replays/hooks/useReplayLayout.tsx [hunk 1]
```diff
@@ -66,9 +69,17 @@ function useActiveReplayTab() {
...
     ),
     setLayout: useCallback(
-      (value: string) =>
-        isLayout(value) ? setParamValue(value) : setParamValue(defaultLayout),
-      [defaultLayout, setParamValue]
+      (value: string) => {
+        const chosenLayout = isLayout(value) ? value : defaultLayout;
+
+        setParamValue(chosenLayout);
+        trackAdvancedAnalyticsEvent('replay-details.layout-changed', {
```
Comment (billyvg) [inline static/app/utils/replays/hooks/useReplayLayout.tsx:76]: Should we have a single "namespace"? e.g. `replay.details-viewed`, `replay.layout-changed` (details is kind of redundant here, but could include it to be consistent: `replay.details-layout-changed`).

File: static/app/utils/analytics/replayAnalyticsEvents.tsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import type {LayoutKey} from 'sentry/utils/replays/hooks/useReplayLayout';
+
+export type ReplayEventParameters = {
+  'replay-details.layout-changed': {
```
Comment (billyvg) [inline static/app/utils/analytics/replayAnalyticsEvents.tsx:4]: Should these be enums? They are used in multiple places. -- Will tsc catch if we change the event name when calling `trackAnalytics...`?

**Review by billyvg — COMMENTED** [review]
I think another use metric to track is when recording is played.

File: static/app/utils/analytics/replayAnalyticsEvents.tsx (hunk 2 above)
Comment (ryan953) [reply static/app/utils/analytics/replayAnalyticsEvents.tsx:4]: The types here are really good actually. TSC matches the key with the args that it expects at the callsite to do the tracking.

File: static/app/utils/replays/hooks/useReplayLayout.tsx (hunk 1 above)
Comment (ryan953) [reply static/app/utils/replays/hooks/useReplayLayout.tsx:76]: Good idea. It's the `replayAnalyticsEvents.tsx` file, so replay as a single prefix makes sense.

**Comment by ryan953** [comment]
> I think another use metric to track is when recording is played.

There's a growing list of things to track in the ticket. https://github.com/getsentry/sentry/issues/38315
I can add play/pause tracking now and get to the rest in followups

**Comment by ryan953** [comment]
> I think another use metric to track is when recording is played.

Actually, i think tracking this opens up a bunch of other things to track inside ReplayController and ReplayContext:
- isEnded
- used next-breadcrumb button
- changed playback speed
- did go fullscreen
probably more.

I want to defer these to the next PR. If this is an example then we can make tickets for the followups and have folks work on them all in paralle.

File: static/app/utils/replays/hooks/useTimeout.tsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
+import {useCallback, useRef} from 'react';
+
+type Options = {
+  onTimeout: () => void;
+  timeMs: number;
+};
+
+function useTimeout({timeMs, onTimeout}: Options) {
```
Comment (billyvg) [inline static/app/utils/replays/hooks/useTimeout.tsx:8]: This should probably be moved to `app/utils`

**Review by billyvg — APPROVED** [review]

**Pull request merged**